### ISSUE TITLE
Add parser support for MySQL's DATE_ADD and DATE_SUB functions

### ISSUE THREAD

**Issue by ghoshrahul** [jOOQ/jOOQ#11452]
### Expected behavior 
On passing a MYSQL query with a **date_add**, it is supposed to execute as expected.

### Actual behavior
It's throwing a Parser Exception

### Steps to reproduce the problem

```java
String sqlQuery = " select users.id, users.created, ifnull(date_add(users.created, interval 2 hour), '0000-00-00 00:00:00') as TEST from users"
ResultQuery jooqQuery = DSL.using(dslContext.configuration()).parser().parseResultQuery(sqlQuery);
```

**Result**:

```
org.jooq.impl.ParserException: Unknown function: [1:184] ...ers.created as User Created, ifnull(date_add([*]users.created, interval 2 hour), '0000-00-00 00:00:00') as TEST from user...
	at org.jooq.impl.ParserContext.exception(ParserImpl.java:11047)
	at org.jooq.impl.ParserImpl.parseUnaryOps(ParserImpl.java:5736)
	at org.jooq.impl.ParserImpl.parseExp(ParserImpl.java:5703)
	at org.jooq.impl.ParserImpl.parseFactor(ParserImpl.java:5680)
	at org.jooq.impl.ParserImpl.parseSum(ParserImpl.java:5632)
	at org.jooq.impl.ParserImpl.parseNumericOp(ParserImpl.java:5617)
	at org.jooq.impl.ParserImpl.parseCollated(ParserImpl.java:5598)
	at org.jooq.impl.ParserImpl.parseConcat(ParserImpl.java:5588)
	at org.jooq.impl.ParserImpl.parsePredicate(ParserImpl.java:4618)
	at org.jooq.impl.ParserImpl.parseNot(ParserImpl.java:4584)
	at org.jooq.impl.ParserImpl.parseAnd(ParserImpl.java:4574)
	at org.jooq.impl.ParserImpl.parseOr(ParserImpl.java:4565)
	at org.jooq.impl.ParserImpl.parseField(ParserImpl.java:5505)
	at org.jooq.impl.ParserImpl.parseField(ParserImpl.java:5428)
	at org.jooq.impl.ParserImpl.parseFieldIfnullIf(ParserImpl.java:8198)
	at org.jooq.impl.ParserImpl.parseTerm(ParserImpl.java:6008)
	at org.jooq.impl.ParserImpl.parseUnaryOps(ParserImpl.java:5723)
	at org.jooq.impl.ParserImpl.parseExp(ParserImpl.java:5703)
	at org.jooq.impl.ParserImpl.parseFactor(ParserImpl.java:5680)
	at org.jooq.impl.ParserImpl.parseSum(ParserImpl.java:5632)
	at org.jooq.impl.ParserImpl.parseNumericOp(ParserImpl.java:5617)
	at org.jooq.impl.ParserImpl.parseCollated(ParserImpl.java:5598)
	at org.jooq.impl.ParserImpl.parseConcat(ParserImpl.java:5588)
	at org.jooq.impl.ParserImpl.parsePredicate(ParserImpl.java:4618)
	at org.jooq.impl.ParserImpl.parseNot(ParserImpl.java:4584)
	at org.jooq.impl.ParserImpl.parseAnd(ParserImpl.java:4574)
	at org.jooq.impl.ParserImpl.parseOr(ParserImpl.java:4565)
	at org.jooq.impl.ParserImpl.parseField(ParserImpl.java:5505)
	at org.jooq.impl.ParserImpl.parseField(ParserImpl.java:5428)
	at org.jooq.impl.ParserImpl.parseSelectList(ParserImpl.java:5351)
	at org.jooq.impl.ParserImpl.parseQueryPrimary(ParserImpl.java:1288)
	at org.jooq.impl.ParserImpl.parseQueryTerm(ParserImpl.java:1211)
	at org.jooq.impl.ParserImpl.parseQueryExpressionBody(ParserImpl.java:1182)
	at org.jooq.impl.ParserImpl.parseSelect(ParserImpl.java:1050)
	at org.jooq.impl.ParserImpl.parseSelect(ParserImpl.java:1042)
	at org.jooq.impl.ParserImpl.parseQuery(ParserImpl.java:919)
	at org.jooq.impl.ParserImpl.parseResultQuery(ParserImpl.java:660)
	at org.jooq.impl.ParserImpl.parseResultQuery(ParserImpl.java:654)
``` 
### Versions

- jOOQ: 3.13
- Java: 11
- Database (include vendor): MYSQL 5.7
- OS: Ubuntu 18.04
- JDBC Driver: com.mysql.jdbc.Driver

**Comment by lukaseder** [comment]
Thanks a lot for your report. Indeed, we don't support parsing that function yet, only the SQL Server style `DATEADD()` function. I'll look into this for the next release.

**Comment by lukaseder** [comment]
This will include supporting `DATE_SUB()` as well

**Comment by lukaseder** [comment]
The reason we hadn't supported this yet, in the past, is because we might not know the data type of the `DATE_ADD()` function (`DATE` or `DATETIME`), and as such might not parse it correctly, unless:

- The first argument is clearly one of the two types, e.g. because it's a literal
- We can look up the first argument from meta data

Both options are now present. If there's still a reason why we can't know the types, then let's assume `DATETIME` semantics.

**Comment by lukaseder** [comment]
Note we do not yet support dynamic interval expressions in jOOQ yet: https://github.com/jOOQ/jOOQ/issues/8792, so this will fix only usages where an interval literal (as in your example) is being used.

**Comment by lukaseder** [comment]
Implemented for jOOQ 3.15. Thanks again for your report.
